// chai-5.0.3/lib/chai/utils/index.js
/*!
 * chai
 * Copyright(c) 2011 Jake Luer <jake@alogicalparadox.com>
 * MIT Licensed
 */

/*!
 * Dependencies that are used for multiple exports are required here only once
 */

import * as checkError from '../../../../../node_modules/check-error-2.0.0/index.js';

/*!
 * test utility
 */

export {test} from './test.js';

/*!
 * type utility
 */

export {type} from './type-detect.js';

/*!
 * expectTypes utility
 */
export {expectTypes} from './expectTypes.js';

/*!
 * message utility
 */

export {getMessage} from './getMessage.js';

/*!
 * actual utility
 */

export {getActual} from './getActual.js';

/*!
 * Inspect util
 */

export {inspect} from './inspect.js';

/*!
 * Object Display util
 */

export {objDisplay} from './objDisplay.js';

/*!
 * Flag utility
 */

export {flag} from './flag.js';

/*!
 * Flag transferring utility
 */

export {transferFlags} from './transferFlags.js';

/*!
 * Deep equal utility
 */

export {default as eql} from '../../../../../node_modules/deep-eql-5.0.1/index.js';

/*!
 * Deep path info
 */

export {getPathInfo, hasProperty} from '../../../../../node_modules/pathval-2.0.0/index.js';

/*!
 * Function name
 */

export function getName(fn) {
  return fn.name
}

/*!
 * add Property
 */

export {addProperty} from './addProperty.js';

/*!
 * add Method
 */

export {addMethod} from './addMethod.js';

/*!
 * overwrite Property
 */

export {overwriteProperty} from './overwriteProperty.js';

/*!
 * overwrite Method
 */

export {overwriteMethod} from './overwriteMethod.js';

/*!
 * Add a chainable method
 */

export {addChainableMethod} from './addChainableMethod.js';

/*!
 * Overwrite chainable method
 */

export {overwriteChainableMethod} from './overwriteChainableMethod.js';

/*!
 * Compare by inspect method
 */

export {compareByInspect} from './compareByInspect.js';

/*!
 * Get own enumerable property symbols method
 */

export {getOwnEnumerablePropertySymbols} from './getOwnEnumerablePropertySymbols.js';

/*!
 * Get own enumerable properties method
 */

export {getOwnEnumerableProperties} from './getOwnEnumerableProperties.js';

/*!
 * Checks error against a given set of criteria
 */

export {checkError};

/*!
 * Proxify util
 */

export {proxify} from './proxify.js';

/*!
 * addLengthGuard util
 */

export {addLengthGuard} from './addLengthGuard.js';

/*!
 * isProxyEnabled helper
 */

export {isProxyEnabled} from './isProxyEnabled.js';

/*!
 * isNaN method
 */

export {isNaN} from './isNaN.js';

/*!
 * getOperator method
 */

export {getOperator} from './getOperator.js';
